src/Command/DrupalUpdate/CheckInfo.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+class CheckInfo
+{
+  /**
+   * @var array
+   */
+  public  $info_files = [];
+
+  /**
+   * @var array
+   */
+  public  $info_details_packages = [];
+
+  /**
+   * @var string[]
+   */
+  public  $packageinfo_key = [
+        'name',
+        'description',
+        'package',
+        'version',
+        'core',
+        'project',
+    ];
+
+  /**
+   * @var array
+   */
+  public $available_updates = [];
+
+  /**
+   * Flag for drupal core update only single time.
+   * Get updates only single time of core for all core modules, themes, profile.
+   * @var bool
+   */
+  public $is_core_get = FALSE;
+
+  /**
+   * @var UpdateScriptUtility
+   */
+  private $updateScriptUtility;
+
+  /**
+   * CheckInfo constructor.
+   */
+  public function __construct() {
+    $this->updateScriptUtility = new UpdateScriptUtility();
+  }
+
+  /**
+   * @param $files1
+   * @param $dir
+   */
+  public function getFilesInfo($scanned_file_path, $dir) {
+    foreach($scanned_file_path as $c_dir){
+      $tm_path = $dir . "/" . $c_dir;
+      if(is_dir($tm_path)){
+        $scanned_file_path_c = array_diff(scandir($tm_path), ['.', '..']);
+        $this->getFilesInfo($scanned_file_path_c, $tm_path);
+      }elseif($this->endsWith($c_dir, '.info')){
+        if(isset($this->info_files[$c_dir])){
+          if(is_array($this->info_files[$c_dir])){
+            $this->info_files[$c_dir][] = $dir . "/" . $c_dir;
+          }else{
+            $temp_p = $this->info_files[$c_dir];
+            $this->info_files[$c_dir]=[];
+            $this->info_files[$c_dir][]=$temp_p;
+            $this->info_files[$c_dir][] = $dir . "/" . $c_dir;
+          }
+        }else{
+          $this->info_files[$c_dir]=$dir . "/" . $c_dir;
+        }
+      }
+    }
+  }
+
+  /**
+   * @param $filepath
+   * @param $package
+   */
+  public function fileGetInfo($filepath, $package) {
+    if (!defined('VERSION')) {
+      define('VERSION', getcwd());
+    }
+    $info_extention_file    = file_get_contents($filepath);
+    $rows = explode("\n", $info_extention_file);
```

Comment:
I think you just need to parse the contents using https://www.php.net/manual/en/function.parse-ini-file.php